kernel/syscall.c

```diff
@@ -6304,6 +6304,8 @@ static long _syscall(void* args_)
         case SYS_myst_oe_free_attestation_certificate:
         case SYS_myst_oe_verify_attestation_certificate:
         case SYS_myst_oe_result_str:
+        case SYS_myst_oe_get_enclave_start_address:
```

Comment:
Do we need to expose these two information (we didn't expose the image_base before)?

Reply:
This is an internal undocumented syscall that exposes it only for the sake of testing.

Reply:
Another option is getting the value via __myst_kernel_args. We already set the start address now (image_data), see https://github.com/deislabs/mystikos/blob/ab55ee086dd2b687488dc34637602c8d6325c39e/tools/myst/enc/enc.c#L515 . We could add additional member to store the base address.

Reply:
I think it is okay. It provides us what we need. :)

Reply:
Yes, I like the idea. Let me see if it works out.

Reply:
I looked into this and we also have to support the Linux target and other TEEs in the future. The base-address/start-address is an SGX-only concept and exposing these to the kernel would break our separation of TEE from kernel. We added the extended syscalls only for the SGX TEE. These calls are not available in the Linux target at all.